.circleci/config.yml

```
@@ -1,43 +1,117 @@
-        pip install -r requirements/base.txt -q
-        pip install -r requirements/test.txt -q
+        # First run the same pipeline as Read-The-Docs
+        # apt-get update && apt-get install -y cmake
+        # using: https://hub.docker.com/r/readthedocs/build
+        # we need to use py3.7 ot higher becase of an issue with metaclass inheritence
+        pyenv global 3.7.3
+        python --version
+        pip install -r requirements/docs.txt
+        cd docs; make clean; make html --debug --jobs 2 SPHINXOPTS="-W"
 
-  tests: &tests
+  checkout_ml_testing: &checkout_ml_testing
    run:
-     name: Testing
+     name: Checkout ml-testing-accelerators
      command: |
-       python --version ; pip --version ; pip list
-       python -m pytest pytorch_lightning -v --junitxml=test-reports/pytest_junit.xml  --ignore=pytorch_lightning/loggers/comet.py --ignore=pytorch_lightning/loggers/mlflow.py --ignore=pytorch_lightning/loggers/neptune.py --ignore=pytorch_lightning/loggers/test_tube.py --ignore=pytorch_lightning/loggers/wandb.py --ignore=pytorch_lightning/metrics/sklearns.py
-     no_output_timeout: 15m
+       git clone https://github.com/GoogleCloudPlatform/ml-testing-accelerators.git
+       cd ml-testing-accelerators
+       git fetch origin 5e88ac24f631c27045e62f0e8d5dfcf34e425e25:stable
+       git checkout stable
+       cd ..
 
-  make_docs: &make_docs
+  build_push_docker: &build_push_docker
+    run:
+      name: Build and push Docker image
+      command: |
+        gcloud --quiet auth configure-docker
+        cd dockers/tpu-tests
+        # TODO: How to find the GITHUB_REF in CircleCI?
+        # $CI_PULL_REQUEST seems to be of form: https://github.com/org/repo-name/pull/11.
+        # Grab the last bit, e.g. pull/11, convert to pull/11/head, and use it
+        # for the GITHUB_REF so Docker can pull the latest pending code in PR.
+        git_ref=$(echo "$CI_PULL_REQUEST" | sed "s/.*pytorch-lightning\///")/head
+        docker build --tag "$GCR_IMAGE_PATH:$CIRCLE_WORKFLOW_JOB_ID" -f Dockerfile --build-arg "TEST_IMAGE=1" --build-arg "GITHUB_REF=$git_ref" .
+        #docker build --tag "$IMAGE:$GITHUB_RUN_ID" -f Dockerfile --build-arg "GITHUB_REF=$GITHUB_REF" --build-arg "TEST_IMAGE=1" .
+        docker push "$GCR_IMAGE_PATH:$CIRCLE_WORKFLOW_JOB_ID"
+
+  deploy_cluster: &deploy_cluster
    run:
-     name: Make Documentation
+     name: Deploy the job on the kubernetes cluster
      command: |
-       # First run the same pipeline as Read-The-Docs
-       # apt-get update && apt-get install -y cmake
-       # using: https://hub.docker.com/r/readthedocs/build
-       # we need to use py3.7 ot higher becase of an issue with metaclass inheritence
-       pyenv global 3.7.3
-       python --version
-       pip install -r requirements/docs.txt
-       cd docs; make clean; make html --debug --jobs 2 SPHINXOPTS="-W"
+       go get github.com/google/go-jsonnet/cmd/jsonnet && \
```

```suggestion
       go get github.com/google/go-jsonnet/cmd/jsonnet
```
I quess this is not needed...

---

Not needed, I see you dropped in a follow up commit, thanks!

---

just some, the remaining seems to be needed...